code/modules/reagents/reagent_containers/blood_pack.dm

```diff
@@ -154,28 +157,39 @@
 
 /obj/item/reagent_containers/blood/examine(mob/user, distance = 2)
 	if (..() && vampire_marks)
-		to_chat(user, "<span class='warning'>There are teeth marks on it.</span>")
+		to_chat(user, SPAN_WARNING("There are teeth marks on it."))
 	return
 
 /obj/item/reagent_containers/blood/attackby(obj/item/P as obj, mob/user as mob)
 	..()
 	if (P.ispen())
 		if (REAGENT_VOLUME(reagents, /decl/reagent/blood) && name != "empty blood pack") //Stops people mucking with bloodpacks that are filled
-			to_chat(usr, "<span class='notice'>You can't relabel [name] until it is empty!</span>")
+			to_chat(user, SPAN_NOTICE("You can't relabel [name] until it is empty!"))
+			return
+		var/blood_name = input(user, "What blood type would you like to label it as?", "Blood Types") in list("A+", "A-", "B+", "B-", "O+", "O-", "AB+", "AB-", "Saline Plus", "Clear", "Cancel")
+		if(blood_name == "Cancel")
 			return
-		var/blood_name = input(usr, "What blood type would you like to label it as?", "Blood Types") in list("A+", "A-", "B+", "B-", "O+", "O-", "AB+", "AB-", "Saline Plus", "Cancel")
-		if (blood_name == "Cancel") return
-		var/obj/item/i = usr.get_active_hand()
-		if (!i.ispen() || !in_range(user, src)) return //Checks to see if pen is still held or bloodback is in range
-		name = "blood pack [blood_name]"
+		var/obj/item/i = user.get_active_hand()
+		if(!i.ispen() || !in_range(user, src)) //Checks to see if pen is still held or bloodback is in range
+			return
+		if(blood_name == "Clear")
+			blood_type = null
+			name = initial(name)
+			desc = initial(desc)
+			to_chat(user, SPAN_NOTICE("You clear the blood pack label."))
+			update_icon()
+			return
+		blood_type = blood_name
```

Comment:
I get why the new var is named `blood_type` for the icons, and this is very tidy, but couldn't you just rename all instances of the `blood_name` var to `blood_type`  instead of doing this? It still works fine, just seems like it'd be more readable in code to use one var.

Reply:
The problem with doing that is with the "Clear" and "Cancel" options for the input thing. With ``blood_name`` those options return without changing much, while if it's ``blood_type``, it gets set as that option and thus clearing the label overlay if the bloodpack is updated

Reply:
Ah yeah, fair enough. Ignore this then.